### PR TITLE
Setup Home Page

### PR DESCRIPTION
**Pull request by peter-gerhard**
@lauraluiz pls review

### CODE REVIEW

**Comment by lauraluiz** [comment]
:+1:  :)

**Comment by peter-gerhard** [comment]
@lauraluiz can you take a quick look again, thx

File: conf/routes [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # ~~~~
 
 # Home page
-GET        /                                                @controllers.HomeController.index()
+GET        /                                                @controllers.ApplicationController.index()
```
Comment (lauraluiz) [inline conf/routes]: This feels weird at first, but on the second thought you can then change to `HomeController` when you don't care about the setup widget and so you improve a bit your performance.

**Comment by lauraluiz** [comment]
:+1:

**Comment by peter-gerhard** [comment]
@schleichardt could you review the last two commits please.
And in case you are able to make use of travis error, I would be happy to know :)

**Comment by schleichardt** [comment]
Yes, you broke the Setup widget. The original index action is required to be in a controller which has no dependencies to the sphere client (e.g. extends directly play.mvc.Controller) since the credentials are not yet present.

File: pt/controllers/ApplicationControllerTest.java [hunk 2]
```diff
@@ -19,9 +19,7 @@
 
     @Test
     public void homeIsAlive() {
-        final Result index = app.injector().instanceOf(ApplicationController.class).index().get(100);
+        final Result index = app.injector().instanceOf(ApplicationController.class).index().get(1000);
         assertThat(index.status()).isEqualTo(200);
-        assertThat(contentAsString(index)).contains("Sunrise Home");
```
Comment (schleichardt) [inline pt/controllers/ApplicationControllerTest.java]: There should be at least one contentual assertion like checking the title.

File: product-catalog/test/productcatalog/pages/ProductDataFactoryTest.java [hunk 3]
```diff
@@ -45,6 +45,10 @@ public void create() {
 
     private ReverseRouter reverseRouter() {
         return new ReverseRouter() {
+            @Override
```
Comment (schleichardt) [inline product-catalog/test/productcatalog/pages/ProductDataFactoryTest.java]: In the test scope we need one default implementation which implements all routes  with null, so we don't need to override every single test every route.

File: app/controllers/HomeController.java [hunk 4]
```diff
@@ -1,29 +1,42 @@
...
-        this.injector = injector;
-        this.setupController = setupController;
+    public HomeController(final ControllerDependency controllerDependency) {
+        super(controllerDependency);
     }
 
-    public F.Promise<Result> index() {
-        return setupController.handleOrFallback(() -> injector.instanceOf(ApplicationController.class).index());
+    public F.Promise<Result> show() {
+        final String language = context().project().languages().stream().findFirst().map(Locale::toLanguageTag)
```
Comment (schleichardt) [inline app/controllers/HomeController.java]: Can you use https://www.playframework.com/documentation/2.4.x/api/java/play/mvc/Http.RequestHeader.html#acceptLanguages-- to get the preferred language of the user?

File: app/pages/ReverseRouterImpl.java [hunk 5]
```diff
@@ -3,17 +3,21 @@
 import common.pages.ReverseRouter;
 import io.sphere.sdk.models.Base;
 import play.mvc.Call;
-import productcatalog.controllers.routes;
```
Comment (schleichardt) [inline app/pages/ReverseRouterImpl.java]: :+1:

File: app/controllers/HomeController.java [hunk 6]
```diff
@@ -1,29 +1,38 @@
...
-    public HomeController(final Injector injector, final SetupController setupController) {
-        this.injector = injector;
-        this.setupController = setupController;
+    public HomeController(final ControllerDependency controllerDependency) {
+        super(controllerDependency);
     }
 
-    public F.Promise<Result> index() {
-        return setupController.handleOrFallback(() -> injector.instanceOf(ApplicationController.class).index());
+    public F.Promise<Result> show() {
```
Comment (schleichardt) [inline app/controllers/HomeController.java]: Not necessary now to change, but this action could be just a 

`public Result show() {`

**Comment by schleichardt** [comment]
:+1:

File: app/controllers/HomeController.java (hunk 6 above)
Comment (peter-gerhard) [inline app/controllers/HomeController.java]: problem was the return type of 
```
public F.Promise<Result> index() {
    return setupController.handleOrFallback(() -> injector.instanceOf(ApplicationController.class).index());
}
```
but yea, we could wrap it in a promise there

**Pull request merged**